bluetooth_mesh/messages/sensor.py

```diff
@@ -0,0 +1,214 @@
+    SENSOR_CADENCE_SET = 0x55
+    SENSOR_CADENCE_SET_UNACKNOWLEDGED = 0x56
+    SENSOR_CADENCE_STATUS = 0x57
+    SENSOR_SETTINGS_GET = 0x8235
+    SENSOR_SETTINGS_STATUS = 0x58
+    SENSOR_SETTING_GET = 0x8236
+    SENSOR_SETTING_SET = 0x59
+    SENSOR_SETTING_SET_UNACKNOWLEDGED = 0x5A
+    SENSOR_SETTING_STATUS = 0x5B
+
+
+SensorGetMinimal = Struct()
+
+SensorGetOptional = Struct(
+    "property_id" / Int16ul
+)
+
+SensorGet = Select(
+    SensorGetOptional,
+    SensorGetMinimal
+)
+
+SensorSettingsGet = Struct(
+    "sensor_property_id" / Int16ul
+)
+
+SensorSettingGet = Struct(
+    Embedded(SensorSettingsGet),
+    "sensor_setting_property_id" / Int16ul
+)
+
+SensorSettingSet = Struct(
+    Embedded(SensorSettingGet),
+    "sensor_setting_raw" / PropertyValue,
+)
+
+SensorSettingStatus = Struct(
+    Embedded(SensorSettingGet),
+    "sensor_setting_access" / Int8ul,
+    "sensor_setting_raw" / PropertyValue,
+)
+
+SensorSettingsStatus = Struct(
+    Embedded(SensorSettingsGet),
+    "sensor_setting_property_ids" / GreedyRange(Int16ul)
+)
+
+SensorDescriptor = Struct(
+    "sensor_property_id" / Int16ul,
+    *DoubleKeyIndex("sensor_negative_tolerance", "sensor_positive_tolerance"),
+    "sensor_sampling_funcion" / Int8ul,
+    "sensor_measurement_period" / Int8ul,
+    "sensor_update_interval" / Int8ul
+)
+
+SensorDescriptorStatus = GreedyRange(
+    Select(
+        SensorDescriptor,
+        Struct("sensor_property_id" / Int16ul)
+    ),
+)
+
+MarshalledPropertyIDShort = EmbeddedBitStruct(
+    "_",
+    "sensor_setting_property_id" / BitsInteger(11),
+    "length" / ExprAdapter(BitsInteger(4), obj_ + 1, obj_ - 1),
+    "format" / Const(0, BitsInteger(1)),
+    reversed=True
+)
+
+MarshalledPropertyIDLong = EmbeddedBitStruct(
+    "_",
+    "sensor_setting_property_id" / BitsInteger(16),
+    "length" / ExprAdapter(BitsInteger(7), obj_ + 1, obj_ - 1),
+    "format" / Const(1, BitsInteger(1)),
+    reversed=True
+)
+
+SensorData = Select(
+    Struct(
+        *MarshalledPropertyIDShort,
+        "sensor_setting_raw" / PropertyValue,
+    ),
+    Struct(
+        *MarshalledPropertyIDLong,
+        "sensor_setting_raw" / Array(this.length, Byte),
+    )
+)
+
+SensorStatus = GreedyRange(
+    SensorData
+)
+
+# TODO: message not implemented due to somewhat complicated structure and lack of examples
+# SensorColumnGet = Struct(
+#     Embedded(SensorSettingsGet),
+#     "raw_value_x" / PropertyValue
+# )
+
+# TODO: message not implemented due to somewhat complicated structure and lack of examples
+# SensorSeriesGet = Struct(
+#     Embedded(SensorSettingsGet),
+#     "raw_value_y" / PropertyValue
+# )
+
+FastCadencePeriodDivisorAndTriggerType = EmbeddedBitStruct(
+    "status_trigger_type" / BitsInteger(1),
+    "fast_cadence_period_divisor" / BitsInteger(7)
+)
+
+UnitlessTriggerDelta = DefaultCountValidator(Int8ul, rounding=1, resolution=0.1)
+
+TriggerDelta = Struct(
+    Switch(
+        this.status_trigger_type,
```

Comment:
Make this an enum (using `EnumAdapter`)?

Reply:
Not sur if that would be easier. 
status_trigger_type is taken from FastCadencePeriodDivisorAndTriggerType field which is an EmbeddedBitStruct on the previous octet.